src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -2303,3 +2303,19 @@ class DropTriggerStatementSegment(BaseSegment):
         Ref("TableReferenceSegment"),
         OneOf("CASCADE", "RESTRICT", optional=True),
     )
+
+
+@postgres_dialect.segment(replace=True)
+class InsertStatementSegment(BaseSegment):
+    """An `INSERT` statement."""
+
+    type = "insert_statement"
+    match_grammar = StartsWith("INSERT")
+    parse_grammar = Sequence(
+        "INSERT",
```

Comment:
Appreciate this isn't necessarily what you were going for in this PR, but if you fancy it, could you take a stab at replicating the entire Postgres insert grammar? If you don't want to do this, can you either have a TODO comment, or comment in the docstring that it is an incomplete representation of the grammar?

Reply:
Looking at the docs, the complete INSERT grammar looks like a bit of a beast so let's maybe finish this PR to address the related issue and then I can look at replicating the whole thing as a separate PR 😄  I've added in a note in the docstring that it is not a complete implementation.